evalml/pipelines/components/transformers/imputers/knn_imputer.py

```diff
@@ -95,23 +98,18 @@ def transform(self, X, y=None):
         X_t = self._component_obj.transform(X[self._cols_to_impute])
         X_t = pd.DataFrame(X_t, columns=self._cols_to_impute)
 
-        # Get Woodwork types for the imputed data
-        new_schema = original_schema.get_subset_schema(self._cols_to_impute)
```

Comment:
No specific reason. I'm now realizing how similar the knn and simple imputers' transform methods really are. Really the only difference is the `bool` logic in the simple imputer to return early if needed.

I can refactor them to follow the same flow of logic and have more similar variable names so that it's easier to read through them.

Reply:
Implemented in 705748238 - this uncovered an undesired behavior when using only `original_schema` to reinit woodwork when fully null columns were present, so I stuck with the behavior of defining the new schema as its own variable but changed its name to be `imputed_schema` to hopefully be slightly more clear than `new_schema` was